daemon/logger/loggerutils/logfile_test.go

```diff
@@ -244,6 +246,77 @@ func TestFollowLogsProducerGone(t *testing.T) {
 	}
 }
 
+type lineDecoder struct {
+	r          *bufio.Reader
+	resetCount int
+}
+
+func (d *lineDecoder) Decode() (*logger.Message, error) {
+	line, err := d.r.ReadString('\n')
+	if err != nil {
+		return nil, err
+	}
+	m := logger.NewMessage()
+	m.Line = []byte(line)
+	return m, nil
+}
+
+func (d *lineDecoder) Reset(r io.Reader) {
+	d.r = bufio.NewReader(r)
+	d.resetCount++
+}
+
+func (d *lineDecoder) Close() {
+}
+
+func TestFollowLogsHandleDecodeErr(t *testing.T) {
```

Comment:
Looks like this test is flaky (seen two failures today 😅 )

Reply:
Nah, let me take a look...

Reply:
I haven't looked closely yet myself, but thought I'd at least leave a comment (thanks!)